api/classes/utils/wp-functions.php

```diff
@@ -0,0 +1,88 @@
+<?php
```

Comment:
I think we could actually simply `require_once( 'wp-includes/functions.php' );` instead of duplicating these, couldn't we? I'm also thinking not just about duplication, but if we end up using this in a WP-loaded context that we'd end up with `function already exists` errors.

Reply:
>  if we end up using this in a WP-loaded context

I don't think this work would ever be used in a WP-loaded context. The goal of this branch is to be as lean as possible — to reduce queries and code down to the bare necessities. If we used it in a WordPress context we'd negate this.

Reply:
> I think we could actually simply `require_once( 'wp-includes/functions.php' );`

I'm wary of including any WordPress files because they will create messy dependency issues. For example, if we include `functions.php` we immediately get warnings about `ABSPATH` and `WPINC`. We won't need many of these functions so I'd prefer to keep it ultra-lean, even if that means a verbatim copy of a function or two.